### PR TITLE
Open Preview extension/Content Sync

### PR DESCRIPTION
**Pull request by kathmbeck**
## Purpose:
Adding Open Preview button to gatsby portfolio starter. The button is now located in the document actions menu in studio, and applies to all content types. As part of this PR, we are following the steps for manually setting up the button on pre existing sites:

https://www.notion.so/gatsbyjs/Gatsby-Open-Preview-Sanity-46fdf6b7b8e343ffaf5306bcba73375b

## Technical details:
We are utilizing the document actions extension to add a new action from https://www.npmjs.com/package/sanity-plugin-gatsby-cloud-preview. The action will redirect folks to a new tab with a content sync url configured to redirect them to a preview of the page they are editing. We have bumped the version of gatsby-source-sanity that utilizes the new PINC builds and content sync functionality.

### CODE REVIEW

File: template/web/gatsby-config.js [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
     'gatsby-plugin-react-helmet',
+    'gatsby-plugin-image',
     {
       resolve: 'gatsby-source-sanity',
       options: {
         ...clientConfig.sanity,
         token,
         watchMode: !isProd,
-        overlayDrafts: !isProd && token
+        overlayDrafts: !isProd || previewEnabled,
```
Comment (DanielSLew) [inline template/web/gatsby-config.js:17]: do we still need to check if `token` exists here, or is that just old code not really doing anything

File: template/web/package.json [hunk 2]
```diff
@@ -15,11 +15,12 @@
     "@sanity/block-content-to-react": "^2.0.7",
     "@sanity/image-url": "^0.140.18",
     "date-fns": "^1.30.1",
-    "gatsby": "^3.0.4",
+    "gatsby": "^4.4.0-next.0",
```
Comment (DanielSLew) [inline template/web/package.json:18]: we probably want a stable version of gatsby here rather than @next

Comment (kathmbeck) [reply template/web/package.json]: Good call, updated this

File: template/web/gatsby-config.js (hunk 1 above)
Comment (kathmbeck) [reply template/web/gatsby-config.js:23]: I used the recommended config in the source plugin readme, but I think that would make sense. If there is no token included, the preview breaks for drafted content.

**Review by DanielSLew — APPROVED** [review]

File: template/web/gatsby-config.js [hunk 3]
```diff
@@ -7,18 +7,20 @@ const clientConfig = require('./client-config')
...
     'gatsby-plugin-react-helmet',
+    'gatsby-plugin-image',
     {
       resolve: 'gatsby-source-sanity',
       options: {
         ...clientConfig.sanity,
         token,
-        watchMode: !isProd,
-        overlayDrafts: !isProd && token
+        watchMode: !previewEnabled,
```
Comment (hdoro) [inline template/web/gatsby-config.js:22]: @kathmbeck with this value for `watchMode`, it'll be used in builds, which could lead to some issues... Could you change this to `!isProd && !previewEnabled`? Thanks 🙇‍♂️

**Review by hdoro — CHANGES_REQUESTED** [review]
@kathmbeck so sorry for the wait on this - thank you for the time you took in writing these changes 🙇‍♂️

There's a tiny issue with the configuration, but other than that the code is good to go. I do need to take a look at the Notion document for activating previews, though. A few things are unclear by now:

1. Don't we need to provide any configuration for the site in `template/studio/resolveDocumentActions.js`?
1. Wouldn't this be counter-intuitive for users coming from sanity.io/create and deploying their sites to Netlify?
  1. Unfortunately we don't currently have the bandwidth to add Cloud as one of our one-click deploy providers. Until we get to it, I feel like it's better if we provide clear docs on how to switch to it, previews included?
  1. We do have plans to make these templates local-first and decoupled from hosting providers, in which case it'd be a no brainer to suggest users to build & host in Gatsby Cloud

**Pull request closed without merging**